todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsViewModel.java

```diff
@@ -16,72 +16,55 @@
 
+
 import android.support.annotation.NonNull;
 import android.support.v4.util.Pair;
 
 import com.example.android.architecture.blueprints.todoapp.data.Task;
 import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
 import com.example.android.architecture.blueprints.todoapp.util.EspressoIdlingResource;
-import com.example.android.architecture.blueprints.todoapp.util.schedulers.BaseSchedulerProvider;
 
 import java.util.List;
 
 import rx.Observable;
-import rx.Subscription;
 import rx.functions.Action0;
-import rx.functions.Action1;
 import rx.functions.Func1;
 import rx.functions.Func2;
-import rx.subscriptions.CompositeSubscription;
+import rx.subjects.PublishSubject;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
  * Listens to user actions from the UI ({@link StatisticsFragment}), retrieves the data and updates
  * the UI as required.
  */
-public class StatisticsPresenter implements StatisticsContract.Presenter {
+public class StatisticsViewModel {
 
     @NonNull
     private final TasksRepository mTasksRepository;
 
     @NonNull
-    private final StatisticsContract.View mStatisticsView;
-
-    @NonNull
-    private final BaseSchedulerProvider mSchedulerProvider;
-
-    @NonNull
-    private CompositeSubscription mSubscriptions;
+    private PublishSubject<Boolean> mProgressIndicatorSubject;
 
-    public StatisticsPresenter(@NonNull TasksRepository tasksRepository,
-                               @NonNull StatisticsContract.View statisticsView,
-                               @NonNull BaseSchedulerProvider schedulerProvider) {
+    public StatisticsViewModel(@NonNull TasksRepository tasksRepository) {
         mTasksRepository = checkNotNull(tasksRepository, "tasksRepository cannot be null");
-        mStatisticsView = checkNotNull(statisticsView, "statisticsView cannot be null!");
-        mSchedulerProvider = checkNotNull(schedulerProvider, "schedulerProvider cannot be null");
 
-        mSubscriptions = new CompositeSubscription();
-        mStatisticsView.setPresenter(this);
+        mProgressIndicatorSubject = PublishSubject.create();
     }
 
-    @Override
-    public void subscribe() {
-        loadStatistics();
-    }
-
-    @Override
-    public void unsubscribe() {
-        mSubscriptions.clear();
-    }
-
-    private void loadStatistics() {
-        mStatisticsView.setProgressIndicator(true);
+    /**
+     * @return A stream of statistics data: <active tasks, completed tasks> pair.
+     */
+    @NonNull
+    public Observable<Pair<Integer, Integer>> getStatistics() {
 
         // The network request might be handled in a different thread so make sure Espresso knows
         // that the app is busy until the response is handled.
         EspressoIdlingResource.increment(); // App is busy until further notice
 
+        // the progress indicator should be visible
+        mProgressIndicatorSubject.onNext(true);
```

Comment:
Imagine someone calls ``getStatistics`` method but doesn't subscribe to it. 
Still, then the progress will be set to true even thou the observable did not start processing.
Setting progress to true should happen most likely on subscription to the observable